aspnetcore/performance/caching/middleware.md

```diff
@@ -50,6 +49,8 @@ Response Caching Middleware only caches server responses that result in a 200 (O
 > [!WARNING]
 > Responses containing content for authenticated clients must be marked as not cacheable to prevent the middleware from storing and serving those responses. See [Conditions for caching](#conditions-for-caching) for details on how the middleware determines if a response is cacheable.
 
+The preceding code typically won't return a cached value to browser. Use [Fiddler](https://www.telerik.com/fiddler), [Postman](https://www.getpostman.com/), or other tool that can explicitly set request headers and are preferred for testing caching. For more information, see [Troubleshooting](#troubleshooting) in this article.
+
 ## Options
 
 Response caching options are shown in the following table.
```

Comment:
I just noticed the size limit on the cache is 100MB. I feel like we might be hitting that, but how can I find out? Does the middleware log warnings?

Reply:
"Overcapacity compaction triggered" is logged at Debug level by `Microsoft.Extensions.Caching.Memory.MemoryCache`. [Source](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs#L359).

Reply:
This comment thread can be resolved @Rick-Anderson, unless you think it's worth mentioning that Response Caching uses IMemoryCache, update docs on Memory Caching to say this log message is written, and then refer to that from here as a way to know if you're hitting limits. That could have value; would have saved me a few mins.

Reply:
Turns out the response caching middleware spawns its own `MemoryCache` instance but doesn't pass it a logger so it gets the null logger and you don't get anything coming out. At least I think that's what's happening. 

https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs#L39-L54

Reply:
> Turns out the response caching middleware spawns its own `MemoryCache` instance but doesn't pass it a logger factory so it gets a no-op logger and you don't get anything coming out. At least I think that's what's happening.
> 
> https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs#L39-L54

@benaadams is that true?

Reply:
In the interest of not hijacking this PR we can discuss this on the [issue I created in aspnetcore](https://github.com/dotnet/aspnetcore/issues/39300). I've proposed a change and if that goes through I'll make sure to open a new PR to document the logging.